keps/sig-ui/5008-kep-template/README.md

```diff
@@ -0,0 +1,893 @@
+  a set of plugins that the project thinks are generally useful for all users
+  in order to provide a rich and relevant Kubernetes user experience.
+* Headlamp Base does not include any plugins. It only includes features that
+  the project deems universally desired in a Kubernetes UI. This is designed
+  for builders looking to have a solid base on which to extend the user
+  experience. They would generally create custom plugins to customize the user
+  experience, include any existing plugins they desire and then theme the UI to
+  fit their branding.
+
+Each version is available as a desktop experience and an in-cluster experience.
+For the default build, the set of plugins differs between the in-cluster and
+desktop installations as there are different user profiles for each environment.
+For example, the plugin that allows managing plugins is not included in the
+in-cluster environment as the cluster admin generally determines which plugins
+to provide to the cluster users.
+
+### Feature highlights
+
+Headlamp has a number of features that we think are interesting for users and
+developers/vendors. Some of those are:
+* A robust plugin system for customization and integrations;
+* Supports both desktop and in-cluster modes;
+* Under very active development, with dedicated resources assigned to it
+  in addition to the active external contributors;
+* Built on a very common technology stack (Typescript, React, Material UI,
+  Vite, Go)
+* Multi-cluster support
+* Graph view, displaying a relational map of a clusters
+* Access and easy set up of community plugins
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+With this proposal we aim to provide the Kubernetes project with the following.
+* A tool to more easily onboard new and beginner-level users
+* A UI that serves as a base for vendors to build upon
+* A modern UI developer environment
+* A feature-rich UI
+* A desktop user experience
+* An integration point for other CNCF and external projects, especially those
+  without a UI focus
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+This proposal does not propose to force people who are using the current
+Dashboard to move to Headlamp.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+The proposal is to ship the Headlamp experience as a new Kubernetes Desktop
+and Kubernetes in-cluster UI.
+
+This means the Headlamp project will become a part of the Kubernetes project
```

Comment:
I think @kubernetes/steering-committee should be involved in this decision.

Do you intend to follow K8s release cycles or be an independent, out-of-tree project?

Reply:
Can we merge as _provisional_ (and maybe iterate) before taking it to Steering?

Reply:
sure

Reply:
After a quick skim I don't understand well if this is a donation of a repository or try to open a new project, repository rules are well defined here https://github.com/kubernetes/community/blob/master/github-management/kubernetes-repositories.md#core-repositories

Can you please clarify what is the goal here?

Reply:
> Do you intend to follow K8s release cycles or be an independent, out-of-tree project?

We may add support for new APIs / resource kinds as they are released, but it would not follow release cycles. i.e. releases are independent in that regard.

Reply:
> Can you please clarify what is the goal here?

The goal is to have a _Kubernetes Desktop_ app to serve as a great desktop experience for Kubernetes users, including novice and experienced, which also serves a base for other CNCF projects to built extensions on.

This proposal involves moving the headlamp-k8s/headlamp repo to Kubernetes, from which we would build the Kubernetes Desktop app. Whether the repo is to be under the official _kubernetes_ org on Github or the _kubernetes-sig_ org, is something I hope someone else can clarify. We are hoping to make this an official desktop app for Kubernetes, so initially we were thinking this meant that the repo could be under the _kubernetes_ org, but the dynamics of where repos should officially live is not very clear to me.

Reply:
It would be in kubernetes-sigs (this is not a "core component that will live in every cluster" or at least I strongly doubt sig-architecture would disagree). Please see the linked documentation for the repo donation process. 

This KEP is not the process to approve a new repo https://github.com/kubernetes/community/blob/master/github-management/kubernetes-repositories.md#core-repositories

Reply:
(Note: kubernetes-sigs is very much official, the separate orgs is a bit of a historical thing, but we do have pretty clearly written rules and provess for this)